src/backend/commands/event_trigger.c

```diff
@@ -1895,7 +1896,16 @@ pg_event_trigger_ddl_commands(PG_FUNCTION_ARGS)
 					 * rather than failing.  This can happen for example with
 					 * some subcommand combinations of ALTER TABLE.
 					 */
-					identity = getObjectIdentity(&addr, true);
+					if(cache_look_ddl_event_trigger_hook)
+					{
+						char temp_identity;
+						(*cache_look_ddl_event_trigger_hook)(&addr, &temp_identity);
+						identity = &temp_identity;
```

Comment:
No need to pass by reference, we can simply return the `identity ` value using hook.